Update/16128 add attribute types documentation

AdamDTak2112 opened the pull request:
## Description
<!-- Please describe what you have changed or added -->
Fixes #16128 . Added accepted attribute types for the Blocks API, including a link to the Wordpress REST API documentation for clarification. 

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
Ran changes in docs through Visual Studio Code's markdown preview testing.

Changes should not affect any other code, as it was purely an addition to documentation. No existing documentation was altered.

## Screenshots <!-- if applicable -->


## Types of changes
<!-- What types of changes does your code introduce?  -->
Only a change in documentation, as clarification was requested through #16128 .

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://wordpress.org/gutenberg/handbook/designers-developers/ -->

AdamDTak2112 committed:
Add attribute validation info to documentation

Added valid inputs to the type field in attributions on block-attributes.md

AdamDTak2112 committed:
added link to WP REST API docs

Soean reviewed (approved):
Thanks for contributing.

mkaz commented on docs/designers-developers/developers/block-api/block-attributes.md:
```diff
@@ -32,6 +32,20 @@ _Example_: Extract the `src` attribute from an image found in the block's markup
+* integer
+	- Matches any number with a zero fractional part
+
+Additional details are found in the documentation for Wordpress's REST API [here](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/).
```
Here's a suggestion for improving the link:

```suggestion
See [Wordpress's REST API documentation](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/) for additional details.
```

chrisvanpatten replied:
Also, capital P :)

```suggestion
See [WordPress's REST API documentation](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/) for additional details.
```

mkaz replied:
hah, how'd I miss that? :-)

mkaz reviewed (approved):
Looks good, I added one suggestion for the link.

Also, not a big deal, but you have a definition for integer, but not number. I would add for both or neither, just the types seems sufficient to me.

chrisvanpatten reviewed (changes requested):
Just adding a block on the PR until this change (a combo of mine + @mkaz's changes) is merged in, then I'd love to see this landed. Thanks!

```suggestion
See [WordPress's REST API documentation](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/) for additional details.
```

AdamDTak2112 committed:
Update docs/designers-developers/developers/block-api/block-attributes.md

Co-Authored-By: Chris Van Patten <hello@chrisvanpatten.com>

chrisvanpatten commented on docs/designers-developers/developers/block-api/block-attributes.md:
````diff
@@ -32,6 +32,20 @@ _Example_: Extract the `src` attribute from an image found in the block's markup
 }
 // { "url": "https://lorempixel.com/1200/800/" }
 ```
+#### `attribute` Type Validation
````
```suggestion

#### `attribute` Type Validation
```

AdamDTak2112 committed:
Updated to delete integer description

I was advised by the core contributors to either add a definition for number, or delete the definition for integer. After noting that a definition of number (either an integer or floating point number) could create more confusion, I opted for the latter choice.

AdamDTak2112 committed:
Update docs/designers-developers/developers/block-api/block-attributes.md

Co-Authored-By: Chris Van Patten <hello@chrisvanpatten.com>

AdamDTak2112 commented:
I've taken the suggestions from @mkaz and @chrisvanpatten. I committed the changes, and please let me know if there is anything else that needs to be change.

Thank you so much for letting me contribute to this! I promise I'll capitalize the P from now on :)

Soean commented:
Thanks for the changes and welcome as a new Gutenberg contributor!
